web/src/components/SpanDetail/SpanDetail.tsx

```diff
@@ -1,25 +1,41 @@
 import {capitalize} from 'lodash';
 import {useCallback} from 'react';
-import {SemanticGroupNames, SemanticGroupNamesToText} from '../../constants/SemanticGroupNames.constants';
-import SpanService from '../../services/Span.service';
-import {TSpan, TSpanFlatAttribute} from '../../types/Span.types';
+
+import {useAssertionForm} from 'components/AssertionForm/AssertionFormProvider';
+import {CompareOperator} from 'constants/Operator.constants';
+import {SemanticGroupNames, SemanticGroupNamesToText} from 'constants/SemanticGroupNames.constants';
+import {useAppSelector} from 'redux/hooks';
+import AssertionSelectors from 'selectors/Assertion.selectors';
+import SpanService from 'services/Span.service';
+import OperatorService from 'services/Operator.service';
+import {TSpan, TSpanFlatAttribute} from 'types/Span.types';
 import Generic from './components/Generic';
 import Http from './components/Http';
 import SpanHeader from './SpanHeader';
 import * as S from './SpanDetail.styled';
-import {useAssertionForm} from '../AssertionForm/AssertionFormProvider';
-import {CompareOperator} from '../../constants/Operator.constants';
-import OperatorService from '../../services/Operator.service';
 
-export interface TSpanDetailProps {
-  testId?: string;
+export interface ISpanDetailsComponentProps {
+  assertions?: IResultAssertions;
+  onCreateAssertion(attribute: TSpanFlatAttribute): void;
   span?: TSpan;
-  resultId?: string;
 }
 
-export interface TSpanDetailsComponentProps {
+export interface IResultAssertions {
```

Comment:
Let's move this to the Assertion types, and instead of using [key: string] I think its better to use Record<>

Reply:
Done